runelite-client/src/main/java/net/runelite/client/plugins/microbot/GirdyScripts/cannonballsmelter/CannonballSmelterScript.java

```diff
@@ -0,0 +1,219 @@
+        if(!Rs2Inventory.hasItem("ammo mould")) {
+            if(!Rs2Bank.isOpen()) {
+                Rs2Bank.openBank();
+            }
+            sleepUntil(() -> Rs2Bank.isOpen());
+            if(!Rs2Bank.hasItem("ammo mould")) {
+                Microbot.showMessage("Could not find ammo mould in bank, exiting...");
+                sleep(3000, 5000);
+                shutdown();
+            }
+            Rs2Bank.withdrawOne("ammo mould");
+            sleepUntil(() -> required(), 3000);
+        }
+        if(!Rs2Bank.hasItem("steel bar")) {
+            Microbot.showMessage("Can't find Steel bars in bank, exiting...");
+            sleep(3000,5000);
+            shutdown();
+        }
+        Rs2Bank.withdrawAll(2353);
+        sleepUntil(() -> Rs2Inventory.hasItem("steel bar"));
+        if(Rs2Inventory.hasItem("steel bar")) {
+            Rs2Keyboard.keyPress(KeyEvent.VK_ESCAPE);
+        }
+        if (!Rs2Inventory.hasItem("steel bar")) {
+            Microbot.showMessage("Could not find item in bank.");
+            shutdown();
+        }
+    }
+
+    public void cannonballAntiBan() {
+        Rs2AntibanSettings.antibanEnabled = true;
+        Rs2AntibanSettings.usePlayStyle = false;
+        Rs2AntibanSettings.randomIntervals = false;
+        Rs2AntibanSettings.simulateFatigue = true;
+        Rs2AntibanSettings.simulateAttentionSpan = true;
+        Rs2AntibanSettings.behavioralVariability = true;
+        Rs2AntibanSettings.nonLinearIntervals = true;
+        Rs2AntibanSettings.profileSwitching = true;
+        Rs2AntibanSettings.timeOfDayAdjust = false;
+        Rs2AntibanSettings.simulateMistakes = true;
+        Rs2AntibanSettings.moveMouseRandomly = true;
+        Rs2AntibanSettings.naturalMouse = true;
+        Rs2AntibanSettings.contextualVariability = true;
+        Rs2AntibanSettings.dynamicIntensity = true;
+        Rs2AntibanSettings.dynamicActivity = false;
+        Rs2AntibanSettings.devDebug = false;
+        Rs2AntibanSettings.takeMicroBreaks = true;
+        Rs2AntibanSettings.playSchedule = false;
+        Rs2AntibanSettings.universalAntiban = false;
+        Rs2AntibanSettings.microBreakDurationLow = 2;
+        Rs2AntibanSettings.microBreakDurationHigh = 10;
+        Rs2AntibanSettings.actionCooldownChance = 1.00;
+        Rs2AntibanSettings.microBreakChance = 0.15;
+        Rs2Antiban.setActivity(Activity.GENERAL_SMITHING);
+    }
+
+    public void mouseOff() {
+        int horizontal = random.nextBoolean() ? -1 : client.getCanvasWidth() + 1;
+        int vertical = random.nextBoolean() ? -1 : client.getCanvasHeight() + 1;
+
+        boolean exitHorizontally = random.nextBoolean();
+        if (exitHorizontally) {
+            Microbot.naturalMouse.moveTo(horizontal, random.nextInt(0, client.getCanvasHeight() + 1));
+        } else {
+            Microbot.naturalMouse.moveTo(random.nextInt(0, client.getCanvasWidth() + 1), vertical);
+        }
+    }
+
+    public void shutdown() {
```

Comment:
You can @Override the shutdown method and call

super.shutdown()

Everything except fo the resetAntiBanSettings is inside the shutdown of the script.java class, so you don't have to duplicate this code.


`@Override
public void shutdown() {
     super.shutdown();
    Rs2Antiban.resetAntibanSettings();
}`